### DISCUSSION TITLE
Add cross-platform logic for file system change journal

### DISCUSSION THREAD

**Discussion by David-Maisonave** [AvaloniaUI/Avalonia#11001] [Q&A] [answered]
**Is your feature request related to a problem? Please describe.**
Add cross-platform **ReadDirectoryChangesW** type API's which can be used to get a list of files changed since a specific time.

**Describe the solution you'd like**
[fswatch](https://github.com/emcrisostomo/fswatch) has this type of cross-platform logic. I would like to see **fswatch** code or something similar, added to Avalonia.

**Describe alternatives you've considered**
The alternative would be to scan the entire path or file system, and comparing every file time stamp to the desired time.  This takes far longer to execute, especially if the path has millions of files.

**Comment by maxkatz6** [comment, marked as answer]
I don't think it's a right place for this request. Avalonia is GUI framework with some minimal extensions.

Can't you use https://learn.microsoft.com/en-us/dotnet/api/system.io.filesystemwatcher?view=net-7.0 ?

**Comment by David-Maisonave** [comment]
Thanks for the link.  I'll see if I can get that to work cross-platform (Win, Mac, & Linux).
